compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -609,6 +610,178 @@ setMatmulVectorDistributionConfig(IREE::GPU::TargetAttr target,
                                                targetSubgroupSize, configDict);
 }
 
+static LogicalResult
+setAttentionVectorDistributionConfig(IREE::GPU::TargetAttr target,
+                                     mlir::FunctionOpInterface entryPoint,
+                                     IREE::LinalgExt::AttentionOp op) {
+  if (target.getWgp().getMma().empty())
+    return failure();
+
+  const int64_t targetSubgroupSize = target.getPreferredSubgroupSize();
+
+  // Get iteration domain bounds.
+  OpBuilder b(op);
+  SmallVector<Range> itDomain = op.getIterationDomain(b);
```

Comment:
This cannot be called here because it constructs IR. I would add an `ExtraClassDeclaration` on the AttentionOp called `getIterationBounds` or something that returns static upper bounds and call that here instead.

Reply:
good idea :)

Reply:
Just curious, seems like in IREE we use `getIterationBounds` as opposed to upstream style of `getStaticLoopRanges`?

Reply:
done! (I think)